src/media/local-audio-track.ts

```diff
@@ -0,0 +1,89 @@
+import { logger } from '../util/logger';
+import { LocalTrack } from './local-track';
+
+export type AudioDeviceConstraints = {
+  deviceId?: string;
+};
+
+export type StaticAudioEncoderConfig = {
+  low: '16kHz';
+};
+
+export type AudioEncoderConfig = {
+  bitrate?: number;
+  sampleRate?: number;
+  sampleSize?: number;
+  channelCount?: number;
+};
+
+export type MicrophoneConstraints = {
+  echoCancellation?: boolean;
+  autoGainControl?: boolean;
+  noiseSuppression?: boolean;
+  microphoneDeviceId?: string;
+  encoderConfig?: AudioEncoderConfig;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type TrackEffect = any;
+/**
+ * Represents a audio track for a audio source.
+ */
+export class LocalAudioTrack extends LocalTrack {
+  private effects: Map<string, TrackEffect> = new Map();
+
+  /**
+   * Get an effect by name.
+   *
+   * @param name - The effect name.
+   * @returns A MicrophoneEffect.
+   */
+  getEffect(name: string): TrackEffect | undefined {
```

Comment:
why do we need copies of getEffect() and addEffect() in LocalAudioTrack() and LocalVideoTrack()? They are basically the same code, the only difference is that addEffect for audio gets the audio track from the stream, while for video gets the video track, but this can be taken care of by just having a getUnderlyingTrack() function in the effect as each effect will only make sense for either audio or video so the effect will know which track to return, or another way would be to just call `getTracks()[0]` because the underlying stream will always have just 1 track (audio or video), right?

Reply:
the `effects` map looks like it could be `protected` in `LocalTrack` as well, and certainly `getEffect` could be there.

Reply:
ya we could move it there but the main reason i moved it to save the effects in respective class files but does not matter can change

Reply:
done